vscode/microsoft-kiota/src/extension.ts

```diff
@@ -706,9 +706,12 @@ async function showUpgradeWarningMessage(clientPath: string, context: vscode.Ext
   }
   const workspaceFileData = await vscode.workspace.fs.readFile(vscode.Uri.file(workspaceFilePath));
   const workspaceFile = JSON.parse(workspaceFileData.toString()) as { kiotaVersion: string };
-  const clientVersion = workspaceFile.kiotaVersion.toLocaleLowerCase();
-  if (clientVersion.toLocaleLowerCase() !== kiotaVersion) {
-    await vscode.window.showWarningMessage(vscode.l10n.t("Client will be upgraded from version {0} to {1}, upgrade your dependencies", clientVersion, kiotaVersion));
+  // don't fail if kiotaVersion isn't in the workspace config file
+  if (workspaceFile.kiotaVersion) {
+    const clientVersion = workspaceFile.kiotaVersion.toLocaleLowerCase();
+    if (clientVersion.toLocaleLowerCase() !== kiotaVersion) {
+      await vscode.window.showWarningMessage(vscode.l10n.t("Client will be upgraded from version {0} to {1}, upgrade your dependencies", clientVersion, kiotaVersion));
```

Comment:
What if it's a downgrade?

Reply:
In what scenarios would a downgrade happen?

Reply:
Downgrading from a version that broke things as you await a fix.

Reply:
I'll check in this PR to unblock the update command crash. There will be a 2nd PR which should address this based on PM input

Reply:
Was this message supposed to show up in which scenarios? When downgrading the version using VS Code UI > Kiota > Uninstall > Install Specific Version > Selecting any version, no banner or message is showing up for me